src/connection.ts

```diff
@@ -247,16 +179,18 @@ export class StableWSConnection<
    */
   _buildUrl = (reqID?: string) => {
     const params = {
-      user_id: this.user.id,
-      user_details: this.user,
-      user_token: this.tokenManager.getToken(),
+      user_id: this.client.userID,
+      user_details: this.client._user,
+      user_token: this.client.tokenManager.getToken(),
       server_determines_connection_id: true,
-      device: this.device,
+      device: this.client.options.device,
       client_request_id: reqID,
     };
     const qs = encodeURIComponent(JSON.stringify(params));
-    const token = this.tokenManager.getToken();
-    return `${this.wsBaseURL}/connect?json=${qs}&api_key=${this.apiKey}&authorization=${token}&stream-auth-type=${this.authType}&X-Stream-Client=${this.userAgent}`;
+    const token = this.client.tokenManager.getToken();
```

Comment:
do we really use `token` and `user_token` on this call?

Reply:
good catch, could be an old mistake